realtime_tools/include/realtime_tools/realtime_publisher.hpp

```diff
@@ -196,33 +204,32 @@ class RealtimePublisher
    * \brief Publishing loop (runs in separate thread)
    *
    * This is the main loop for the non-realtime publishing thread. It:
-   * 1. Waits for new messages (State::REALTIME)
+   * 1. Waits for new messages (State::NON_REALTIME)
    * 2. Copies the message data
    * 3. Publishes the message through the ROS publisher
-   * 4. Returns to State::NON_REALTIME to allow realtime updates
+   * 4. Returns to State::REALTIME to allow realtime updates
    *
    * The loop continues until keep_running_ is set to false.
    */
   void publishingLoop()
   {
     is_running_ = true;
-    turn_.store(State::NON_REALTIME, std::memory_order_release);
 
     while (keep_running_) {
       MessageT outgoing;
 
       {
+        turn_.store(State::REALTIME, std::memory_order_release);
```

Comment:
Similar to my comment above, I feel like turn_ should be checked when locked so this line should move under the unique_lock below. Again I could be wrong about this.